<?xml version="1.0" encoding="UTF-8"?>
<root BTCPP_format="4"
      main_tree_to_execute="MainTree">
  <BehaviorTree ID="MainTree">
    <Sequence name="root">
      <Sequence>
        <DecideEnemy now_enemy_id="0"
                     target_enemy_id="{choose_enemy_id}"/>
        <GimbalChooseEnemyNode now_enemy_id="{choose_enemy_id}"/>
      </Sequence>
      <Sequence>
        <TimeCompuLimit now_game_time="0"
                        first_interval="5"
                        time_switch="1"
                        time_limit1="90"
                        time_limit2="15"/>
        <Patrol3Node area_choose="0"
                     time_begin="0"
                     navigation_point="{now_navigation_point}"/>
        <Patrol2Node distance_limit_min="1.0"
                     now_navigation_point="{now_navigation_point}"
                     navigation_point="{now_navigation_point}"
                     self_point="{self_point}"
                     distance_limit_max="2.0"/>
        <NavToSpecificPlace target_place="{now_navigation_point}"/>
      </Sequence>
    </Sequence>
  </BehaviorTree>

  <BehaviorTree ID="Subtree">
    <Sequence>
      <Patrol3Node area_choose="5"
                   time_begin="0"
                   navigation_point="{now_navigation_point}"/>
      <Patrol2Node distance_limit_min="1.0"
                   now_navigation_point="{now_navigation_point}"
                   navigation_point="{now_navigation_point}"
                   self_point="{self_point}"
                   distance_limit_max="2.0"/>
      <NavToSpecificPlace target_place="{now_navigation_point}"/>
    </Sequence>
  </BehaviorTree>

  <!-- Description of Node Models (used by Groot) -->
  <TreeNodesModel>
    <Action ID="DecideEnemy"
            editable="true">
      <input_port name="now_enemy_id"/>
      <output_port name="target_enemy_id"/>
    </Action>
    <Action ID="GimbalChooseEnemyNode"
            editable="true">
      <input_port name="now_enemy_id"/>
    </Action>
    <Action ID="NavToSpecificPlace"
            editable="true">
      <input_port name="target_place"/>
    </Action>
    <Action ID="Patrol2Node"
            editable="true">
      <input_port name="distance_limit_min"/>
      <inout_port name="now_navigation_point"/>
      <input_port name="navigation_point"/>
      <input_port name="self_point"/>
      <input_port name="distance_limit_max"/>
    </Action>
    <Action ID="Patrol3Node"
            editable="true">
      <inout_port name="area_choose"/>
      <input_port name="time_begin"/>
      <output_port name="navigation_point"/>
    </Action>
    <Action ID="TimeCompuLimit"
            editable="true">
      <output_port name="now_game_time"/>
      <input_port name="first_interval"/>
      <input_port name="time_switch"/>
      <input_port name="time_limit1"/>
      <input_port name="time_limit2"/>
    </Action>
  </TreeNodesModel>

</root>
